### PR TITLE
Adds the AGLS-37

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
image
Adds the AGLS. Basically just a remake of my old PR with some modernizations. Lower ROF (1 Second) compared to before, otherwise pretty much the same.
It can rotate now, and HE lands on the tile you click now.
## Why It's Good For The Game
It's unique, cool, and per French stature of Limitations I still have limitation on making this PR considering Lootpacks are done finally.
## Changelog
:cl:
add: Added new AGLS mounted grenade launcher.
/:cl:

### CODE REVIEW

**Comment by novaepee** [comment]
image

**Comment by UsernameIs-UnAvailable** [comment]
Oh boy, more power creep by the One and Only Pariah919.

**Comment by Nicolai904** [comment]
cant wait to get perma knockbacked back into marines

**Comment by DemoCrack** [comment]
Get some.

**Review by Kuro020 — APPROVED** [review]

**Comment by Lumipharon** [comment]
Isn't this like a guaranteed juggle/stagger/slow death to any xeno caught in the open?

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -1645,6 +1645,64 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 /datum/ammo/smoke_burst/do_at_max_range(turf/T, obj/projectile/P)
 	drop_nade(T.density ? P.loc : T)
 
+
+/datum/ammo/ags_shrapnel
+	name = "40mm fragmentation grenade"
```
Comment (XSlayer300) [inline code/modules/projectiles/ammo_datums.dm:1650]:
```suggestion
	name = "fragmentation grenade"
```

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -3321,3 +3379,13 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	name = "smoke grenade shell"
 	nade_type = /obj/item/explosive/grenade/smokebomb
 	icon_state = "smoke_shell"
+
+/datum/ammo/grenade_container/ags_grenade
+	name = "40mm grenade shell"
```
Comment (XSlayer300) [inline code/modules/projectiles/ammo_datums.dm:3384]:
```suggestion
	name = "grenade shell"
```

File: code/modules/projectiles/guns/mounted.dm [hunk 3]
```diff
@@ -447,3 +449,44 @@
 		if(EXPLODE_LIGHT)
 			take_damage(rand(10, 50))
 
+//AGLS-37, or Automatic Grenade Launching System 37, a fully automatic mounted grenade launcher that fires fragmentation and HE shells, can't be turned.
+
+/obj/item/weapon/gun/standard_agls
+	name = "\improper AGLS-37 Kauser automatic grenade launcher"
+	desc = "The AGLS-37 automatic grenade launching IFF capable system, it's too heavy to be wielded or operated without the tripod. On the back, it reads: \"The Explosions and Fragmentation from this weapon ARE NOT friendly fire capable. Kauser is not obligated to buy you new body parts for you or your friends if you lose them.\"\nCan be deployed with Crtl-Click. It CANNOT be turned once deployed, due to a lack of swivels, pick it up to move your cone of fire. Can be repaired with a blowtorch once deployed."
```
Comment (XSlayer300) [inline code/modules/projectiles/guns/mounted.dm:456]:
```suggestion
	desc = "The AGLS-37 automatic grenade launching IFF capable system, it's too heavy to be wielded or operated without the tripod. On the back, it reads: \"The explosions and fragmentation from grenades fired from this weapon ARE NOT IFF capable. Kauser is not obligated to buy you new body parts for you or your friends if you lose them.\"\nCan be deployed with Ctrl-Click. It CANNOT be turned once deployed, due to a lack of swivels, pick it up to move your cone of fire. Can be repaired with a blowtorch once deployed."
```
sooo, does that mean that the AGLS is not capable of FF?

File: code/modules/reqs/supplypacks.dm [hunk 4]
```diff
@@ -189,6 +189,21 @@ WEAPONS
...
+	contains = list(/obj/item/weapon/gun/standard_agls)
+	cost = 70
+
+/datum/supply_packs/weapons/ags_highexplo
+	name = "AGLS-37 AGL high explosive grenades"
+	contains = list(/obj/item/ammo_magazine/standard_agls)
+	cost = 4
+
+/datum/supply_packs/weapons/ags_frag
+	name = "AGLS-37 AGL fragmentation grenades"
```
Comment (XSlayer300) [inline code/modules/reqs/supplypacks.dm:203]:
```suggestion
	name = "AGLS-37 AGL Fragmentation Grenades"
```

File: code/modules/reqs/supplypacks.dm [hunk 5]
```diff
@@ -189,6 +189,21 @@ WEAPONS
 	contains = list(/obj/item/ammo_magazine/tl102)
 	cost = 10
 
+/datum/supply_packs/weapons/ags_emplacement
+	name = "AGLS-37 Mounted Automated Grenade Launcher"
+	contains = list(/obj/item/weapon/gun/standard_agls)
+	cost = 70
+
+/datum/supply_packs/weapons/ags_highexplo
+	name = "AGLS-37 AGL high explosive grenades"
```
Comment (XSlayer300) [inline code/modules/reqs/supplypacks.dm:198]:
```suggestion
	name = "AGLS-37 AGL High Explosive Grenades"
```

**Comment by novaepee** [comment]
image

**Comment by maikilangiolo** [comment]
Alright, after 15 rounds using it, I can say the AGLS is unusable and worthless. Here's why:
- the HE grenades always travel the max distance and don't land where you click
- despite this, they work like normal grenades in instead of exploding on hit. Result is that all your shots fly past xenos (beyond your scope range) and none hit
- second, airburst is bugged. The shrapnel flies through marines without hitting them, and I'm still unsure if it hits xenos at all or also flies through them. 
- airburst also does basically no damage. I hit 5 grenades on a mature sentinel and he walked it off fine. The shrapnel doing a 360° spread means 80% shrap will go in useless directions, and only 2 or 3 actually hit a xeno at best.
- the IFF is worthless because **the scope is 3 tiles** and the grenades are still grenades. Can't use the AGLS at all if marines are remotely close to your target
- a grenade launcher with mini scope does more damage, has a bigger zoom, can move quickly, rotate and fires faster. It's also free and juggle-able.

There needs to be implemented changes for this gun to work.

Honest suggestion that takes like 2 line changes and would work great:
- make the AGLS fire the exact same ammo as the GL54 (the 20mm airburst launcher). 
This would make the AGLS worth it, the ammo is already balanced, and it would essentially make the gun a heavier, larger capacity fixed GL54, but without NV scope. The ammo also has less FF potential, because with only a 3 tile scope, you basically can't use normal grenades unless you're at the front (which the gun prevents you to do)

Otherwise:
- critical: make the HE ammo actually land where you click, not go at max distance and blow up somewhere into the darkness. Why even bother?
- critical: make the airburst ammo release shrapnel in a 45° to 60° cone in the direction of fire, not some stupid 360° radius, unless you increase massively the shrapnel amount. 
- requested: either heavily reduce the deploy and undeploy time, or make the AGLS able to turn. Honestly, I understand the reason why, but this gun is unusable. You can't fire due to FF grenades and 3 tile scope is too short. If you move at the front of the push, you constantly have to wait 6 seconds to even move your weapon, you're vulnerable to damage and can't turn to defend yourself.
- needed: increase the reload time, but provided the gun is actually improved with these suggestions, otherwise the only thing this 70 point junk has got going for it, is fast reload.

**Comment by maikilangiolo** [comment]
Update: sometimes the He grenade range/bounce is inconsistent. It has happened the grenades would drop at my feet despite targeting the furthest area possible (or somehow bouncing back 10 tiles)

**Comment by Pariah919** [comment]
>    the HE grenades always travel the max distance and don't land where you click
    despite this, they work like normal grenades in instead of exploding on hit. Result is that all your shots fly past xenos (beyond your scope range) and none hit
    second, airburst is bugged. The shrapnel flies through marines without hitting them, and I'm still unsure if it hits xenos at all or also flies through them.
    airburst also does basically no damage. I hit 5 grenades on a mature sentinel and he walked it off fine. The shrapnel doing a 360° spread means 80% shrap will go in useless directions, and only 2 or 3 actually hit a xeno at best.
    the IFF is worthless because the scope is 3 tiles and the grenades are still grenades. Can't use the AGLS at all if marines are remotely close to your target
    a grenade launcher with mini scope does more damage, has a bigger zoom, can move quickly, rotate and fires faster. It's also free and juggle-able.
>
Intentional, will probably change since there's a det time now.
didn't in my testing, probably something up with iff flags so i'll check again
that's inherient to how shrapnel works
inherient to how the grenade works
Might increase the scope range to TAT levels.

>equested: either heavily reduce the deploy and undeploy time, or make the AGLS able to turn. Honestly, I understand the reason why, but this gun is unusable. You can't fire due to FF grenades and 3 tile scope is too short. If you move at the front of the push, you constantly have to wait 6 seconds to even move your weapon, you're vulnerable to damage and can't turn to defend yourself.
needed: increase the reload time, but provided the gun is actually improved with these suggestions, otherwise the only thing this 70 point junk has got going for it, is fast reload.
>
no, no and understandable

**Comment by maikilangiolo** [comment]
> Might increase the scope range to TAT levels

Rip. I think this shouldn't be the right change, it will probably turn the gun into OP: TAT scope is marginally shorter than a sniper scope, are you sure you want grenades to be lobbed that far? 

> inherient to how the grenade works

And I take it you will not improve the airburst which does less damage than a holdout (but hey, at least it does 20 damage to everything in an AoE (it will only hit marines)).

Look, it's your PR and you can do what you want with it. I like the gun, it looks good, it sounds good, and has potential. I gave you very accurate feedback, but I must admit it stings that's it all ignored because of the (imo) flawed design of the gun. Getting a sniper scope, while fixing half the issues, will make the gun more unbalanced than addressing the disappoinment that's airburst ammo and the fact you can't aim the HE nades at all (yes, I love being forced to direct hit xenos and have the grenade bounce back into marines and myself).

**Comment by isotope436** [comment]
fun fact this is one of TWO items that has not been adjusted for the ABSOLUTELY POINTLESS inflation pr which means i can do stuff like this at the start of the game (this isnt even a fraction of the abuse thats possible) 
eee

**Comment by isotope436** [comment]
also being unable to rotate this weapon makes it violently useless as it can get flanked with insane ease

**Comment by isotope436** [comment]
make the nades impact or the gun able to rotate or the scope actually a scope and i will consider it good however atm its violently situational and not worth the actual point cost (once it gets fixed) and ends up either one killing itself or two killing a push because you cant see far enough.

also please give it explosive armor so its not its own hard counter please

**Comment by isotope436** [comment]
because i find it funny you should give it to engis :trollface:

**Comment by maikilangiolo** [comment]
After your recent changes, I can give very positive feedback. HE grenades are usable, the swivel is nice. However
- frag rounds are still very underwhelming and extremely dangerous to use due to the massive AoE
- it is possible to add a delay to swiveling, so you can be flanked more easily and you can't quick scope turn around? It's a better option that no swivel at all.
- I'd try if you can remove IFF, and add aim mode with slightly lower rof (idk how much. 0.9? 0.8?), since current ROF with IFF is a bit much, but if it was restricted to non-aim mode it would be better

And can I ask something? For just one or two days, can we try giving the AGLS the same ammo as the GL54? Just to try how it would feel, because I'm thinking it might be a good idea

**Comment by maikilangiolo** [comment]
1. Finally merge this
2. The gun needs bomb armor 
3. Merge this finally

**Comment by isotope436** [comment]
> 

iff is required either that or he increases the scopes view range this thing is already a detriment to pushs if your not VIOLENTLY cautious about where you place rounds the removal of iff means im gonna to down right refuse to buy it as RO because i know its gonna do nothing but kill marines. also it doesnt need a rate of fire decrease the thing only kills you if you sit and tank rounds and is currently the slowest firing gun in the game next to bolt actions (even the svd fires faster than this thing) 

also yes the gun needs bomb armor it still counters itself the xenos dont even need to try to deal with it anymore because it will eventually just magically ping a round off the ground a tile infront of it (some bug or desync issue is my guess) and kill itself happened so many times i stopped using it all together

**Comment by isotope436** [comment]
> After your recent changes, I can give very positive feedback. HE grenades are usable, the swivel is nice. However
> 
> * frag rounds are still very underwhelming and extremely dangerous to use due to the massive AoE
> * it is possible to add a delay to swiveling, so you can be flanked more easily and you can't quick scope turn around? It's a better option that no swivel at all.
> * I'd try if you can remove IFF, and add aim mode with slightly lower rof (idk how much. 0.9? 0.8?), since current ROF with IFF is a bit much, but if it was restricted to non-aim mode it would be better
> 
> And can I ask something? For just one or two days, can we try giving the AGLS the same ammo as the GL54? Just to try how it would feel, because I'm thinking it might be a good idea

iff is required either that or he increases the scopes view range this thing is already a detriment to pushs if your not VIOLENTLY cautious about where you place rounds the removal of iff means im gonna to down right refuse to buy it as RO because i know its gonna do nothing but kill marines. also it doesnt need a rate of fire decrease the thing only kills you if you sit and tank rounds and is currently the slowest firing gun in the game next to bolt actions (even the svd fires faster than this thing)

also yes the gun needs bomb armor it still counters itself the xenos dont even need to try to deal with it anymore because it will eventually just magically ping a round off the ground a tile infront of it (some bug or desync issue is my guess) and kill itself happened so many times i stopped using it all together

**Comment by maikilangiolo** [comment]
> it will eventually just magically ping a round off the ground a tile infront of it (some bug or desync issue is my guess)

Yeah no clue why that happens. Otherwise, I'm afraid I diagnose you with cope and skill issue, simply learn to use the gun better.

**Comment by isotope436** [comment]
the floor impact issue was touched on once before but there wasnt actually anything that was negatively impacted by it (until now) so it was left to rot i forget what caused it but its not an issue if we get full bomb armor which i dont see why we shouldnt theres no suicide xenos so this changes nothing besides making it unable to kill itself [also i dont remember where i found the post since i forgot to save it and i dont feel like digging through discord again]

onto the topic at hand. there is no skill issue or coping involved outside of xenos on a few rounds they got to face it in a siege setting (basically the only thing this fuckers good for outside of drilling a hole through xeno mazes) you want to test gl54 that will inertly do nothing for the weapon outside of probably buffing it and theres really no reason to test it as i havent seen any complaints on the damage more just the fact its a auto gl and the xenomains cant comprehend its exist (average baby man crying that comes with any new marine req weapon)

onto aimmode theres no need to try and give it aimmode if the ROF is an issue as all that would need to be done is tweak the ROF and even then the reason iff is required is because this gun is ment to throw rounds down a long hallway (it goes a solid 10ish tiles past view range maybe more if i remember correctly) and is basically a glorified point and shoot morter which double the cost plus it already takes 10 seconds to place the gun theres really no need to give it more of a cooldown to be useful plus the rounds dont det on impact so xenos usually only get hit with about a 3rd of the rounds (if you catch them lacking) before they skitter off and even then its pretty easy to avoid if your smart enough.

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**